executor/stubs.go

```diff
@@ -43,7 +45,29 @@ func MountStubsCleaner(dir string, mounts []Mount) func() {
 			if st.Size() != 0 {
 				continue
 			}
-			os.Remove(p)
+			// Back up the timestamps of the dir for reproducible builds
+			// https://github.com/moby/buildkit/issues/3148
+			dir := filepath.Dir(p)
```

Comment:
What if the parent dir or the mount also did not exist?

Reply:
Covered in the test for `tmp/foo2/bar`